WMF Framework/RelatedPageExtensions.swift

```diff
@@ -0,0 +1,91 @@
+import CocoaLumberjackSwift
+
+extension WMFArticle {
+    
+    @objc public func update(withRelatedPage page: RelatedPage) {
+        if let thumbnail = page.thumbnail, let url = thumbnail.source {
+            thumbnailURLString = url
+            thumbnailURL = URL(string: url)
+        }
+
+        displayTitleHTML = page.title
+        pageID = NSNumber(value: page.pageId)
+        lastModifiedDate = DateFormatter.wmf_iso8601()?.date(from: page.touched)
+        snippet = page.articleDescription
+        wikidataDescription = page.articleDescription
+    }
+}
+
+extension NSManagedObjectContext {
+    @objc public func wmf_createOrUpdateArticleSummmaries(withRelatedPages relatedPages: [WMFInMemoryURLKey: RelatedPage]) throws -> [WMFInMemoryURLKey: WMFArticle] {
```

Comment:
> This object is now used in the CoreData operations to retrieve article data.
> On this point, I left the older methods that use the summary object since other parts of the app use it. This code is not super DRY, so suggestions are more than welcome

An alternative you could do is convert the RelatedPage objects to ArticleSummary down in RelatedSearchFetcher below. This way, classes outside of the fetcher don't need to change and these extensions in this file aren't necessary.

I'm a bit unfamiliar with these Core Data update methods, so it makes me a little nervous duplicating it and potentially spreading some complicated logic.

I made some commits with what I mean in a `page-related-endpoint-suggestions` branch. They are 0042a0e301026cd5f18f99cd1138c8c1d5fc2dcd, fca9a0272d5e7e556f86f350a22883ec218a6340, 554201efabe0cd34059f83c2f083680ebd961f50, eda1dff1f267b2272e0fd7cdbf09baafcc73eee3.

Reply:
I pushed these commits, per our discussion.